### ISSUE TITLE
Compatibility with worker threads

### ISSUE THREAD

**Issue by yamalight** [vercel/pkg#716]
Is pkg compatible with Worker threads?

I'm trying to package a simple app that uses Worker threads with pkg and it just freezes.
And if I add web server (e.g. fastify) - it seems like it tries to spawn it over and over again instead of running actual worker.

Here's a repo with minimal app that reproduces this behavior: https://github.com/yamalight/pkg-worker-fastify-issue

If you run it as-is in Node - it works perfectly fine.
If you package it and run the resulting binary, it'll go crazy with `Error: listen EADDRINUSE: address already in use 0.0.0.0:8080` even though fastify is initialized only once.
If you remove fastify and repackage it - you'll only see empty output and it'll never finish.

Any ideas why that might happen and how to fix this?

**Comment by saikoux12** [comment]
do you solve this problem?

**Comment by erossignon** [comment]
Is this duplicate #775 ?

**Comment by robertsLando** [comment]
#775 is a duplicate of this, at least this also has a repo to reproduce the error. @yamalight Could you try with latest version?

**Comment by yamalight** [comment]
@robertsLando yep, seems to work as expected now! 🎉

**Comment by robertsLando** [comment]
Thanks for reporting back. Closing as fixed

**Comment by JonathanWoodward** [comment]
@yamalight I have a project similar to this which I am having issues with and am hoping you may know the answer.

@GCN0185 MINGW64 /c/git/SquirrelViewData (master)
$ pkg index.js

pkg@5.5.2
Targets not specified. Assuming:
node16-linux-x64, node16-macos-x64, node16-win-x64

@GCN0185 MINGW64 /c/git/SquirrelViewData (master)
$ node index.js
PRAGMA journal_mode = WAL
Connection to local database has been established successfully.
Server ready at http://localhost:8454/ http://192.168.10.47:8454

@GCN0185 MINGW64 /c/git/SquirrelViewData (master)
$ ./index-win.exe
PRAGMA journal_mode = WAL
Connection to local database has been established successfully.
Server ready at http://localhost:8454/ http://192.168.10.47:8454
[worker eval]:1
[native code]
^^^^

SyntaxError: Unexpected identifier
at new Script (node:vm:101:7)
at createScript (node:vm:259:10)
at Object.runInThisContext (node:vm:307:10)
at node:internal/process/execution:75:19
at [worker eval]-wrapper:6:22
at evalScript (node:internal/process/execution:74:60)
at MessagePort. (node:internal/main/worker_thread:176:7)
at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:562:20)
at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28)
worker exited with code 1
[worker eval]:1
[native code]
`

I believe pkg is having trouble finding the module I require from inside the worker which is why it gets stuck.

`module.exports = WorkerThread = () => {
    const { parentPort } = require('worker_threads');
    const db = require('better-sqlite3')('./store.sqlite', { verbose: console.log });`

The error will go away is I comment out the require('better-sqlite3'), the same thing will occur if I try include any of the node_modules.
